test/e2e/redundancy/gateway_failover.go

```diff
@@ -41,12 +61,24 @@ func testEnginePodRestartScenario(f *subFramework.Framework) {
 	enginePod := f.AwaitSubmarinerEnginePod(framework.ClusterA)
 	By(fmt.Sprintf("Found submariner engine pod %q on %q", enginePod.Name, clusterAName))
 
+	By(fmt.Sprintf("Ensuring that the gateway reports as active %q", clusterAName))
+	activeGateways := f.GetGatewaysWithHAStatus(subv1.HAStatusActive)
+	Expect(activeGateways).To(HaveLen(1))
+
```

Comment:
We should verify that the found active `Gateway`'s name matches that of the gateway node found above.

Reply:
Or better yet, call `AwaitGatewayWithStatus`.

Reply:
But I want to verify there's only "1". I can't do that with Await, I can call Await just before this, which would make sense, also would remove any glitches if it's starts as passive and then moves to active.

Reply:
Ok, I did it in a slightly different way, let me know what you think